pyspeckit/spectrum/units.py

```diff
@@ -968,6 +834,27 @@ def _update_from(self, obj):
                 'velocity_convention', 'wcshead', 'xtype'):
             self.__dict__[attr] = obj.__dict__[attr]
 
+    def find_equivalencies(self, velocity_convention, center_frequency, center_frequency_unit, equivalencies):
+        """
+        Utility function used to validate parameters and add equivalencies
+        """
+        if velocity_convention and center_frequency and equivalencies:
+            raise ValueError('You cannot specify all 3: velocity_convetion, center_frequency and equivalencies') 
+
+        # velocity_convention xor center_frequency
+        if bool(velocity_convention) != bool(center_frequency):
+            raise ValueError('Either specify both velocity_convention and center_frequency, or neither')
+
+        if not hasattr(center_frequency, 'unit'):
+            if center_frequency_unit:
+                center_frequency = center_frequency * u.Unit(center_frequency_unit)
+            else:
+                raise AttributeError("If center_frequency_unit is None, center_frequency should have unit attribute")
+
+        equivalencies = conventions[velocity_convention](center_frequency) + equivalencies
+        return (center_frequency, equivalencies)
```

Comment:
This is good.  If `center_frequency` is not defined, we're in some trouble, but I think that is OK for now because we can't do anything about it until an upstream bug is fixed.